### PR TITLE
Fix MongoDB config validation if replica set is used.

### PR DESCRIPTION
**Pull request by bernd**
It is okay to have an empty mongodb_host if mongodb_replica_set is defined.

Graylog server startup fails without this if a replica set is in use.

Additional whitespace fix.

### CODE REVIEW

**Comment by bernd** [comment]
Forgot to adjust the test. Sorry...

File: graylog2-server/src/main/java/org/graylog2/configuration/MongoDbConfiguration.java [hunk 1]
```diff
@@ -173,11 +173,11 @@ public MongoClientURI getMongoClientURI() {
 
     @ValidatorMethod
     public void validate() throws ValidationException {
-        if((isNullOrEmpty(getHost()) || isNullOrEmpty(getDatabase())) && isNullOrEmpty(getUri())) {
+        if ((isNullOrEmpty(getHost()) || isNullOrEmpty(getDatabase()) || getReplicaSet() == null) && isNullOrEmpty(getUri())) {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/configuration/MongoDbConfiguration.java]: I guess this should be `if ((isNullOrEmpty(getHost()) && getReplicaSet() == null) || isNullOrEmpty(getDatabase()) && isNullOrEmpty(getUri())) {` instead. Bonus points for checking that the replica set is not empty.

Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/configuration/MongoDbConfiguration.java]: You are right, thanks!

**Comment by bernd** [comment]
Updated with fixed condition and test.

**Comment by joschi** [comment]
LGTM.

**Pull request merged**